Android: do not call setuid().

Android runs apps under a seccomp filter that kills the process on setuid()
(SIGSYS) since Android 8, so gnuplot died at startup. Same fix as Termux.

--- a/src/plot.c	2026-07-25 20:28:44.000000000 +0200
+++ b/src/plot.c	2026-10-09 11:33:41.507105340 +0200
@@ -233,7 +233,7 @@
 
 /* make sure that we really have revoked root access, this might happen if
    gnuplot is compiled without vga support but is installed suid by mistake */
-#ifdef __linux__
+#if defined(__linux__) && !defined(__ANDROID__)
     if (setuid(getuid()) != 0) {
 	fprintf(stderr,"gnuplot: refusing to run at elevated privilege\n");
 	exit(EXIT_FAILURE);
